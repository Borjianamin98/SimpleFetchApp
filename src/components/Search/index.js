import React, {Component} from "react";
import PropTypes from "prop-types";

export default class Search extends Component {
    componentDidMount() {
        if (this.input) {
            this.input.focus();
        }
    }

    render() {
        const {value, onChange, onSubmit, children} = this.props;
        return (
            <form onSubmit={onSubmit}>
                <input
                    type="text"
                    value={value}
                    onChange={onChange}
                    ref={instance => this.input = instance}
                />
                <button type="submit">
                    {children}
                </button>
            </form>
        );
    }
}

Search.propTypes = {
    value: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    onSubmit: PropTypes.func.isRequired,
    children: PropTypes.node.isRequired,
}