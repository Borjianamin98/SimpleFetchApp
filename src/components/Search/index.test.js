import Search from "./index.js";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import React from "react";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16.1";

Enzyme.configure({adapter: new Adapter()});

describe('Search', () => {
    const searchButton = <Search
        value="defaultValue"
        onChange={(event) => console.log(event)}
        onSubmit={(event) => console.log(event)}
    >Search</Search>

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(searchButton, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('has a valid snapshot', () => {
        const component = renderer.create(searchButton);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});