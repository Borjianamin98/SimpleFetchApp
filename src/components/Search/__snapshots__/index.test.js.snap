// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search has a valid snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value="defaultValue"
  />
  <button
    type="submit"
  >
    Search
  </button>
</form>
`;
